src/states/node.rs

```diff
@@ -1568,25 +1511,20 @@ impl Node {
         if let Authority::Client { ref client_key, .. } = *dst {
             client_key == self.full_id.public_id().signing_public_key()
         } else {
-            self.is_proper() && self.peer_mgr.routing_table().is_recipient(dst.to_destination())
+            self.is_proper() && self.peer_mgr.routing_table().is_recipient(&dst.to_destination())
         }
     }
 
     /// Returns a `sent_to` entry for the next hop message, and a list of target peer IDs.
     fn get_targets(&self,
                    routing_msg: &RoutingMessage,
                    route: u8,
-                   hop: &XorName,
                    sent_to: &[XorName])
                    -> Result<(Vec<XorName>, Vec<PeerId>), RoutingError> {
         if self.is_proper() {
-            let destination = routing_msg.dst.to_destination();
-            let targets = self.peer_mgr
+            let targets = try!(self.peer_mgr
                 .routing_table()
-                .target_nodes(destination, hop, route as usize)
-                .into_iter()
-                .filter(|target| !sent_to.contains(target.name()))
```

Comment:
I thought it might be useful for debugging for a while, but I'm happy to remove it.  I'll leave it until you confirm one way or the other if you don't mind?